web/source/kmwembedded.ts

```diff
@@ -333,99 +354,33 @@ namespace com.keyman.text {
 
       // Can't just split on '-' because some layers like ctrl-shift contain it.
       let separatorIndex = keyName.lastIndexOf('-');
-      var layer = core.keyboardProcessor.layerId;
+
       if (separatorIndex > 0) {
-        layer = keyName.substring(0, separatorIndex);
         keyName = keyName.substring(separatorIndex+1);
       }
-      if(layer == 'undefined') {
-        layer=core.keyboardProcessor.layerId;
-      }
 
       // Note:  this assumes Lelem is properly attached and has an element interface.
       // Currently true in the Android and iOS apps.
-      var Lelem=keymanweb.domManager.getLastActiveElement(),keyShiftState=com.keyman.text.KeyboardProcessor.getModifierState(layer);
-      
+      var Lelem=keymanweb.domManager.getLastActiveElement();
       keymanweb.domManager.initActiveElement(Lelem);
 
-      var nextLayer: string;
-
       // This should be set if we're within this method... but it's best to guard against nulls here, just in case.
-      if(osk.vkbd.popupBaseKey && osk.vkbd.popupBaseKey['key']) {
-        // This is set with the base key of our current subkey elsewhere within the engine.
-        var baseKey: com.keyman.osk.OSKKeySpec = osk.vkbd.popupBaseKey['key'].spec;
-        var found = false;
-
-        if(baseKey.coreID == keyName) {
-          nextLayer = baseKey.nextlayer;
-          found = true;
-        } else {
-          // Search for the specified subkey so we can retrieve its useful properties.
-          // It should be within the popupBaseKey's subkey list.
-          for(let subKey of baseKey.sk) {
-            if(subKey.coreID == keyName) {
-              // ... to consider:  why are we not just taking the keyspec wholesale right here?
-              nextLayer = subKey.nextlayer;
-              found = true;
-              break;
-            }
-          }
+      if(osk.vkbd.subkeyGesture) {
+        let gesture = osk.vkbd.subkeyGesture as com.keyman.osk.embedded.SubkeyDelegator;
+
+        try {
+          // Can trigger an error if the corresponding subkey cannot be found.
+          gesture.resolve(keyName);
+        } catch (e) {
+          let err = e as Error;
+          // Prevent the Android app from triggering a "fatal error" keyboard reset.
```

Comment:
I don't want to see 'workarounds' like this in library code for problems in consumers of the library, where we control the consumer anyway. It's just silly.

Here's the thing: I think the Android app *should* generate a fatal error keyboard reset for debug (alpha) builds, but be more graceful in release builds. It was excellent today in catching errors when @MakaraSok was testing this PR. 

So: I will turn that into an issue for Android, and we'll aim to address that reasonably soon (should be straightforward - famous last words).